packages/relayer-lib/src/tx_builder/cosmos_to_cosmos.rs

```diff
@@ -0,0 +1,243 @@
+                .into_iter()
+                .filter(|e| match e {
+                    EurekaEvent::SendPacket(se) => {
+                        now >= se.packet.timeoutTimestamp
+                            && se.packet.sourceChannel == target_channel_id
+                    }
+                    _ => false,
+                })
+                .map(|e| async {
+                    match e {
+                        EurekaEvent::SendPacket(se) => {
+                            let ibc_path = se.packet.receipt_commitment_path();
+                            self.source_tm_client
+                                .prove_path(&[b"ibc".to_vec(), ibc_path], target_height)
+                                .await
+                                .map(|(v, p)| {
+                                    if v.is_empty() {
+                                        Some(MsgTimeout {
+                                            packet: Some(se.packet.into()),
+                                            proof_unreceived: p.encode_vec(),
+                                            proof_height: Some(Height {
+                                                revision_number: client_state
+                                                    .latest_height
+                                                    .unwrap_or_default()
+                                                    .revision_number,
+                                                revision_height: target_height.into(),
+                                            }),
+                                            signer: self.signer_address.clone(),
+                                        })
+                                    } else {
+                                        None
+                                    }
+                                })
+                        }
+                        _ => unreachable!(),
+                    }
+                }),
+        )
+        .await?
+        .into_iter()
+        .flatten()
+        .collect::<Vec<_>>();
+
+        let (src_send_events, src_ack_events): (Vec<_>, Vec<_>) = src_events
+            .into_iter()
+            .filter(|e| match e {
+                EurekaEvent::SendPacket(se) => {
+                    se.packet.timeoutTimestamp > now && se.packet.destChannel == target_channel_id
+                }
+                EurekaEvent::WriteAcknowledgement(we) => {
+                    we.packet.sourceChannel == target_channel_id
+                }
+                _ => false,
+            })
+            .partition(|e| match e {
+                EurekaEvent::SendPacket(_) => true,
+                EurekaEvent::WriteAcknowledgement(_) => false,
+                _ => unreachable!(),
+            });
+
+        let _recv_msgs = future::try_join_all(src_send_events.into_iter().map(|e| async {
+            match e {
+                EurekaEvent::SendPacket(se) => {
+                    let ibc_path = se.packet.commitment_path();
```

Comment:
Yes, this whole function is too long. I will refactor this once it is actually working. No need to create an issue I think since I wouldn't forget this as it is very annoying.

Reply:
created an issue https://github.com/cosmos/solidity-ibc-eureka/issues/173